apps/src/sharedFunctionalBlocks.js

```diff
@@ -320,19 +318,141 @@ function installString(blockly, generator) {
  * condition-value pairs before the default value.
  */
 function installCond(blockly, generator, numPairs) {
-  var blockName = 'functional_cond_' + numPairs;
+  // TODO(brent) - rtl
+
+  var blockName = 'functional_cond';
   blockly.Blocks[blockName] = {
     helpUrl: '',
     init: function() {
-      var args = [];
-      for (var i = 0; i < numPairs; i++) {
-        args.push({name: 'COND' + i, type: 'boolean', default: 'false'});
-        args.push({name: 'VALUE' + i, type: 'none', default: ''});
+      this.pairs_ = [];
+      this.setFunctional(true, {
+        headerHeight: 30
+      });
+
+      var options = {
+        fixedSize: { height: 35 }
+      };
+
+      var plusField = new Blockly.FieldIcon('+');
+      plusField.getRootElement().addEventListener('mousedown',
+        _.bind(this.addRow, this));
+
+      this.appendDummyInput()
+        .appendTitle(new Blockly.FieldLabel('cond', options))
+        .setAlign(Blockly.ALIGN_CENTRE);
+
+      this.appendFunctionalInput('DEFAULT');
+
+      this.appendDummyInput('PLUS')
+        .appendTitle(plusField)
+        .setInline(true);
+
+      this.setFunctionalOutput(true);
+
+      while(this.pairs_.length < numPairs) {
+        this.addRow();
+      }
+    },
+
+    /**
+     * Add another condition/value pair to the end.
+     */
+    addRow: function () {
+      // id is either the last value plus 1, or if we have no values yet 0
+      var id = this.pairs_.length > 0 ? (this.pairs_.slice(-1) * 1 + 1) : 0;
+      this.pairs_.push(id);
+
+      var cond = this.appendFunctionalInput('COND' + id);
+      cond.setHSV.apply(cond, functionalBlockUtils.colors.boolean);
+      cond.setCheck('boolean');
+      this.moveInputBefore('COND' + id, 'DEFAULT');
+
+      this.appendFunctionalInput('VALUE' + id)
+        .setInline(true)
+        .setHSV(0, 0, 0.99);
+      this.moveInputBefore('VALUE' + id, 'DEFAULT');
+
+      var minusInput = this.appendDummyInput('MINUS' + id)
+        .setInline(true);
+
+      if (this.pairs_.length > 1) {
+        var minusField = new Blockly.FieldIcon('-');
+        minusField.getRootElement().addEventListener('mousedown',
+          _.bind(this.removeRow, this, id));
+        minusInput.appendTitle(minusField);
+      }
+
+      this.moveInputBefore('MINUS' + id, 'DEFAULT');
+    },
+
+    /**
+     * Remove the condition/value pair with the given id. No-op if no row with
+     * that id.
+     */
+    removeRow: function (id) {
+      var index = this.pairs_.indexOf(id);
+      if (index === -1 || this.pairs_.length === 1) {
```

Comment:
Slightly prefer `_.contains(this.pairs_, index)` or `_(this.pairs_).contains(index)`